### PR TITLE
[Imaging Uploader] Validate filename in front-end before attempting upload

### PR DESCRIPTION
**Pull request by davidblader**
### Brief summary of changes
Checks the filename for candidate uploads before attempting to place it onto the server

### This resolves issue...
attempting to upload incorrect filenames

- [ ] Redmine? #15109

- [ ] Github? #2870 

### To test this change...

- [ ] try uploading a file to imaging uploader where the filename does not match the PSCID_CANDID_VISIT-LABEL convention for a candidate upload. you should get an error swal

### CODE REVIEW

**Comment by driusan** [comment]
@davidblader still failing Travis

File: modules/imaging_uploader/jsx/UploadForm.js [hunk 1]
```diff
@@ -77,16 +77,42 @@ class UploadForm extends React.Component {
...
+      }
+      // Make sure file follows PSCID_CandID_VL[_*].zip|.tgz|.tar.gz format
+      const pcv = data.pSCID + '_' + data.candID + '_' + data.visitLabel;
+      const pcvu = pcv + '_';
+      const properName = new RegExp("^" + pcv + ".(zip|tgz|tar.gz)");
+      const properNameExt = new RegExp("^" + pcvu + ".*(.(zip|tgz|tar.gz))");
+      if (!fileName.match(properName) && !fileName.match(properNameExt)) {
+        swal({
+          title: "File improperly named!",
+          text: "File name must match " + pcv +
```
Comment (cmadjar) [inline modules/imaging_uploader/jsx/UploadForm.js]: I would rephrase the error message to the following:
"File name and values in the PSCID, CandID and Visit label fields of the instrument do not match. Please verify that the information entered in the fields or the file name are correct."

I find it dangerous to suggest them that it is the file name that is not correct (it is possible that what is not correct is one of the field instead of the filename)

**Comment by davidblader** [comment]
@cmadjar i fixed it :~)
